### PR TITLE
Fix control select and exception when moving controls

### PR DESCRIPTION
**Pull request by elachlan**
Related: #8515

I have fixed the exceptions in `DropSourceBehavior` and `SelectionUIService`. But the resize via the editor still doesn't work.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8522)

### CODE REVIEW

**Comment by elachlan** [comment]
@dreddy-work I am unsure what code manages the drag to resize behaviour. So I have fixed what I can for now.

**Comment by dreddy-work** [comment]
> @dreddy-work I am unsure what code manages the drag to resize behavior. So I have fixed what I can for now.

A revised version of the original source code can be found in the designer repository. It has been significantly improved to function properly outside of the VS process for .NET designer. This runtime copy of code is still used by designers who are working outside of Visual Studio.

**Comment by dreddy-work** [comment]
> But the resize via the editor still doesn't work.

Given this was working fine in .NET 6.0, we may need to cross check the changes that went in around `BahviourService `and related code.

**Comment by elachlan** [comment]
Both the resize and designer action menu don't work. I tried bisecting it to track down when it was introduced. But I ran into compile errors pretty quickly. 

Are you sure that the resize operation (using the grip) and action menus worked in .NET 6?

**Comment by dmitrii-drobotov** [comment]
> Are you sure that the resize operation (using the grip) and action menus worked in .NET 6?

@elachlan - It definitely worked in .NET 6, my results when running DemoConsole from the latest commit in `release/6.0` branch:
DemoConsoleNet6

**Comment by RussKie** [comment]
To find the offending commit you should run `git bisect`.

________________________________

**Comment by elachlan** [comment]
Thank you so much!!! I will get a patch sorted asap.

**Comment by elachlan** [comment]
@dmitrii-drobotov @dreddy-work that seems to fix it. Please review.

**Review by dmitrii-drobotov — APPROVED** [review]
Looks good, thank you!

**Review by dreddy-work — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.cs [hunk 1]
```diff
@@ -887,7 +887,7 @@ private void DisableAdorners(IServiceProvider serviceProvider, BehaviorService b
         /// </summary>
         private void InitiateDrag(Point initialMouseLocation, ICollection dragComps)
         {
-            dragObjects = (List<IComponent>)dragComps;
+            dragObjects = dragComps.Cast<IComponent>().ToList();
```
[the remark this answers is not in the document]
Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.cs:890]: The plan is to refactor the whole codebase to remove usage of ArrayList. Tracking issue is #8140

**Comment by elachlan** [comment]
@dreddy-work does the team want to backport this?

**Comment by dreddy-work** [comment]
> @dreddy-work does the team want to backport this?

Given this does not impact VS designer scenarios and .NET6.0 (which has long term support) is not impacted, we wait for the further reports with business justifications and decide if it needs backporting then.

**Pull request merged**